Easydict/Feature/Configuration/NewConfiguration.swift

```diff
@@ -0,0 +1,595 @@
+    }
+
+    func didSetAutoCopyOCRText() {
+        logSettings(["auto_copy_ocr_text": autoCopyOCRText])
+    }
+
+    func didSetAutoCopyFirstTranslatedText() {
+        logSettings(["auto_copy_first_translated_text": autoCopyFirstTranslatedText])
+    }
+
+    func didSetLanguageDetectOptimize() {
+        logSettings(["detect_optimize": languageDetectOptimize])
+    }
+
+    func didSetDefaultTTSServiceType() {
+        if #available(macOS 13, *) {
+            let value = defaultTTSServiceType
+            logSettings(["tts": value])
+        }
+    }
+
+    func didSetShowGoogleQuickLink() {
+        postUpdateQuickLinkButtonNotification()
+
+        EZMenuItemManager.shared().googleItem?.isHidden = !showGoogleQuickLink
+
+        logSettings(["show_google_link": showGoogleQuickLink])
+    }
+
+    func didSetShowEudicQuickLink() {
+        postUpdateQuickLinkButtonNotification()
+
+        EZMenuItemManager.shared().eudicItem?.isHidden = !showEudicQuickLink
+
+        logSettings(["show_eudic_link": showEudicQuickLink])
+    }
+
+    func didSetShowAppleDictionaryQuickLink() {
+        postUpdateQuickLinkButtonNotification()
+
+        EZMenuItemManager.shared().appleDictionaryItem?.isHidden = !showAppleDictionaryQuickLink
+
+        logSettings(["show_apple_dictionary_link": showAppleDictionaryQuickLink])
+    }
+
+    func didSetHideMenuBarIcon() {
+        if !NewAppManager.shared.enable {
+            hideMenuBarIcon(hidden: hideMenuBarIcon)
+        }
+
+        logSettings(["hide_menu_bar_icon": hideMenuBarIcon])
+    }
+
+    func didSetEnableBetaNewApp() {
+        logSettings(["enable_beta_new_app": enableBetaNewApp])
+    }
+
+    func didSetFixedWindowPosition() {
+        logSettings(["show_fixed_window_position": fixedWindowPosition])
+    }
+
+    func didSetMouseSelectTranslateWindowType() {
+        logSettings(["show_mouse_window_type": mouseSelectTranslateWindowType])
+    }
+
+    func didSetShortcutSelectTranslateWindowType() {
+        logSettings(["show_shortcut_window_type": shortcutSelectTranslateWindowType])
+    }
+
+    func didSetAdjustPopButtomOrigin() {
+        logSettings(["adjust_pop_buttom_origin": adjustPopButtomOrigin])
+    }
+
+    func didSetAllowCrashLog() {
+        EZLog.setCrashEnabled(allowCrashLog)
+        logSettings(["allow_crash_log": allowCrashLog])
+    }
+
+    func didSetAllowAnalytics() {
+        logSettings(["allow_analytics": allowAnalytics])
+    }
+
+    func didSetClearInput() {
+        logSettings(["clear_input": clearInput])
+    }
+
+    func didSetFontSizeIndex() {
+        NotificationCenter.default.post(name: .init(ChangeFontSizeView.changeFontSizeNotificationName), object: nil)
+    }
+
+    func didSetAppearance() {
+        DarkModeManager.sharedManager().updateDarkMode()
+    }
+}
+
+// MARK: Window Frame
+
+private extension Configuration {
+    func windowFrameWithType(_ windowType: EZWindowType) -> CGRect {
```

Comment:
Try use `Defaults.Bridge` to comfort `CGRect` to `Serializable` and introduce a Defaults.Key for it?